### DISCUSSION TITLE
Create list from array for Evaluation

### DISCUSSION THREAD

**Discussion by cartyc** [open-policy-agent/community#157] [OPA and Rego] [answered]
Hi!

Have a fun Rego question for the community. I am trying to evaluate a list of JSON array to confirm a specific object has a requited name and haven't figured out how to grab it successfully yet. I was wondering how it might be possible to make a list of the values from the array and then evaluate that list to see if the name exists.

The array looks roughly like this
```
{
"data":[
         {
          "name": "resource",
 		  "resource":{
            "data":{
              "name":"good-eval-name", <-- This is what I want to capture in a list for evaluation
            }
           }
         },
                  {
          "name": "resource",
 		  "resource":{
            "data":{
              "name":"bad-eval-name", <-- This is what I want to capture in a list for evaluation
            }
           }
         }
     ],
}
```

In this case I would be wanting to confirm the resource has a data array with the name `good-eval-name`.

The above example should pass and the below should fail.

```
{
"data":[
         {
          "name": "resource",
 		  "resource":{
            "data":{
              "name":"bad-eval-name", 
            }
           }
         },
                  {
          "name": "resource",
 		  "resource":{
            "data":{
              "name":"also-bad-eval-name", 
            }
           }
         }
     ],
}
```

thanks!

**Comment by anderseknert** [comment, marked as answer]
If you want to aggregate an array of the resource names, you could either use an [array comprehension](https://www.openpolicyagent.org/docs/latest/policy-language/#array-comprehensions), like:

```rego
import future.keywords.in

obj := {"data": [
	{
		"name": "resource",
		"resource": {"data": {"name": "good-eval-name"}},
	},
	{
		"name": "resource",
		"resource": {"data": {"name": "bad-eval-name"}},
	},
]}

resource_names := [name | some item in obj.data; name := item.resource.data.name]
```
Or you could use a [partial rule](https://www.openpolicyagent.org/docs/latest/#partial-rules) to build a set of them:

```rego
resource_names[name] {
    some item in obj.data
    name := item.resource.data.name
}
```
(You could also use a set comprehension to achieve the same outcome, they are functionally equivalent)

Then simply use `in` to check for membership:

```rego
not "bad-eval-name" in resource_names
```

I'd probably choose to work with this as a set, since ordering doesn't seem to matter (and neither uniqueness).

Let me know if that answers your question or if I should elaborate on something :)

**Reply by cartyc** [reply]
Thanks @anderseknert !!! Just gave it a shot and it worked like a charm :)
